coala_quickstart/generation/Utilities.py

```diff
@@ -91,6 +92,23 @@ def split_by_language(project_files):
             for lang in exts[ext]:
                 lang_files[lang.lower()].add(file)
                 lang_files["all"].add(file)
+        elif name and not ext:
+            with open(file, 'r') as data:
+                hashbang = data.readline()
+                if(re.match(
+                        '/(^#![(a-z)|\/]*[ ][a-z]*)|(^#![(a-z)|\/]*)/',
+                        hashbang)):
+                    language = get_language_from_hashbang(hashbang)
+                    try:
+                        for ext in exts:
+                            for lang in exts[ext]:
+                                if language == lang.lower():
+                                    lang_files[lang.lower()].add(file)
```

Comment:
For a very, very, very, very, very tiny performance improvement, use `language` instead of `lang.lower()` here :D